src/excmds.ts

```diff
@@ -428,7 +428,10 @@ export async function guiset_quiet(rule: string, option: string) {
  */
 //#background
 export async function guiset(rule: string, option: string) {
-    await guiset_quiet(rule, option)
+    if (!(await guiset_quiet(rule, option))) {
```

Comment:
I'm not sure this works. guiset_quiet returns the result of Native.write:

https://github.com/tridactyl/tridactyl/blob/ae1d4afc1de888c5c2979671c9fb4caee96f1b6b/src/excmds.ts#L385

And Native.write returns the result of sendNativeMsg:

https://github.com/tridactyl/tridactyl/blob/ae1d4afc1de888c5c2979671c9fb4caee96f1b6b/src/lib/native.ts#L285

Which returns a MessageResp:

https://github.com/tridactyl/tridactyl/blob/ae1d4afc1de888c5c2979671c9fb4caee96f1b6b/src/lib/native.ts#L51

Which has the following type:

https://github.com/tridactyl/tridactyl/blob/ae1d4afc1de888c5c2979671c9fb4caee96f1b6b/src/lib/native.ts#L28-L34

I'm not sure what `code` and `error` are but I think that's what we need to check.

Reply:
https://github.com/tridactyl/tridactyl/blob/9d7b1107cd67abf98a5659a6f36c2cb5c4e14588/src/excmds.ts#L369

This PR is checking that we don't return here, where less subtle errors like "you haven't got the native messenger installed" are caught.

Reply:
It looks like the MessageResp is formed in the handle_message function in native_main.py.  The only property that _could_ be returned in a "write" MessageResp is the cmd key so I am not sure how useful it is to check the other MessageResp property (e.g., write, code).  Please advise.

https://github.com/tridactyl/tridactyl/blob/ae1d4afc1de888c5c2979671c9fb4caee96f1b6b/native/native_main.py#L394-L397

https://github.com/tridactyl/tridactyl/blob/ae1d4afc1de888c5c2979671c9fb4caee96f1b6b/native/native_main.py#L470-L472

Reply:
Ah, indeed, the native messenger's error handling works very differently from what I remember - sorry for the noise!

Reply:
@glacambre No problem, I appreciate your help.